README.md

```diff
@@ -1,115 +1,165 @@
+
+# The Casper Sidecar
+
+- [Summary of Purpose](#summary-of-purpose)
+- [System Components and Architecture](#system-components-and-architecture)
+   - [The SSE server](#the-sse-server)
+   - [The REST API server](#the-rest-api-server)
+	- [The Admin API server](#the-admin-api-server)
+	- [The RPC API server](#the-rpc-api-server)
+- [Configuring the Sidecar](#configuring-the-sidecar)
+	- [RPC server setup](#rpc-server-setup)
+	- [SSE server setup](#sse-server-setup)
+		- [Configuring SSE node connections](#configuring-sse-node-connections)
+		- [Configuring SSE legacy emulations](#configuring-sse-legacy-emulations)
+		- [Configuring the event stream](#configuring-the-event-stream)
+	- [REST server setup](#rest-server-setup)
+	- [Storage setup](#setup-storage)
+	- [Database connectivity setup](#database-connectivity-setup)
+		- [SQLite database](#sqlite-database)
+		- [PostgreSQL database](#postgresql-database)
+	- [Admin server setup](#admin-server-setup)
+- [Running and Testing the Sidecar](#running-and-testing-the-sidecar)
+	- [Prerequisites](#prerequisites)
+   - [Running the Sidecar](#running-the-sidecar)
+   - [Testing the Sidecar](#testing-the-sidecar)
+- [Swagger Documentation](#swagger-documentation)
+- [OpenAPI Specification](#openapi-specification)
+- [Troubleshooting Tips](#troubleshooting-tips)
+	- [Checking liveness](#checking-liveness)
+	- [Checking the node connection](#checking-the-node-connection)
+	- [Diagnosing errors](#diagnosing-errors)
+	- [Monitoring memory consumption](#monitoring-memory-consumption)
+	- [Ensuring sufficient storage](#ensuring-sufficient-storage)
+	- [Inspecting the REST API](#inspecting-the-rest-api)
+	- [Limiting concurrent requests](#limiting-concurrent-requests)
 
 ## Summary of Purpose
 
-The Casper Event Sidecar is an application that runs in tandem with the node process. It's main purpose is to:
-* offload the node from broadcasting SSE events to multiple clients
-* provide client features that aren't part of the nodes' functionality, nor should they be
+The Casper Sidecar is an application running in tandem with the node process. It allows subscribers to monitor a node's event stream, query stored events, and query the node's JSON RPC API, thus receiving faster responses and reducing the load placed on the node. Its primary purpose is to:
+
+* Offload the node from broadcasting SSE events to multiple clients.
+* Provide client features that aren't part of the nodes' functionality, nor should they be.
 
 While the primary use case for the Sidecar application is running alongside the node on the same machine, it can be run remotely if necessary.
 
-### System Components & Architecture
+## System Components and Architecture
+
+The Casper Sidecar provides the following functionalities:
+* A server-sent events (SSE) server with an `/events` endpoint that streams all the events received from all connected nodes. The Sidecar also stores these events.
```

Comment:
It may also provide legacy endpoints, as mentioned later in the docs. Not sure if worth mentioning here, just sayin' :)

Reply:
Good catch! I've added a point for this with commit 5dd23af92487c0ceda14322165e56a6e900d781d.